starters/qwik-graphql-tailwind/src/routes/data-fetching/data-fetching.tsx

```diff
@@ -34,16 +47,27 @@ export const DataFetching = component$(() => {
           value={greetingResource}
           onPending={() => <>Loading...</>}
           onRejected={(error) => <>Error: {error.message}</>}
-          onResolved={(message) => <strong>{message}.</strong>}
+          onResolved={({ data }) => <strong>{data.hello}.</strong>}
         />
       </div>
     </div>
   );
 });
 
-export async function fetchGreeting(greeting: string, abortController?: AbortController): Promise<string> {
-  const resp = await fetch(`https://api.starter.dev/hello?greeting=${greeting}`, {
+export async function fetchGreeting(greeting: string, abortController?: AbortController): Promise<HelloResponse> {
+  if (!greeting) {
+    greeting = 'there';
+  }
+
+  const { executeQuery$ } = useQuery(GET_GREETING);
```

Comment:
Just a question here, the `useQuery` looks great, and I am just curious, since the `fn` passed to `useResource$` will be transformed via `$` and become lazy loaded trunk, in this case `fetchGreeting` will become a `$` function, so do we need to optimize the `executeQuery$` too?

Reply:
Good question! I would say indeed in this case we don't really need that optimization, but it doesn't hurt to keep it generic in case we'd want to use the `useQuery` from a different context....